### PR TITLE
marshallers: corba: remove leading underscore from type names

### LINKED ISSUE #132: orogen cannot handle "C style" typedef'ed structs correctly
I'm almost sure this is a regression because I (we? :P) used this before and I don't remember having to work around it...

Given the definition:
```cpp
typedef struct _Foobar {
    int dummy_field;
} Foobar;
```

It seems Typelib does not generate definitions for `_Foobar` but still creates convertion methods for it:

```
.orogen/typekit/transports/corba/Convertions.cpp:17:34: error: ‘_Foobar’ is not a member of ‘orogen::Corba’
bool toCORBA( orogen::Corba::_Foobar& corba, _Foobar const& value );
^~~~~~~
.orogen/typekit/transports/corba/Convertions.cpp:17:34: note: suggested alternative: ‘Foobar’
bool toCORBA( orogen::Corba::_Foobar& corba, _Foobar const& value );
^~~~~~~
Foobar
.orogen/typekit/transports/corba/Convertions.cpp:17:43: error: ‘corba’ was not declared in this scope
bool toCORBA( orogen::Corba::_Foobar& corba, _Foobar const& value );
^~~~~
``` 

Possible regression? I think I have used this before..

### PR DESCRIPTION
**Pull request by g-arjones**
Fixes #132

### CODE REVIEW

**Comment by doudou** [comment]
I feel funny about this ... it relies on wrong IDL being generated and "normalized" by the IDL-to-C++ compiler

I'd also like the IDL to match the generated code to be honest, but I can also understand that you don't want to go down into the rabbit hole.

**Comment by g-arjones** [comment]
> I'd also like the IDL to match the generated code to be honest, but I can also understand that you don't want to go down into the rabbit hole.

If you mean just the .idl file, I did and was rather easy but has no real effect. I can add that back as well if you think it makes more sense

**Comment by doudou** [comment]
I would do both, yes. The IDL spec rejects having a leading underscore in an identifier, so I believe it would be best to not rely on the omniidl current behavior of ignoring them (instead of having an error). That behavior could change in the future.

**Comment by g-arjones** [comment]
> I would do both, yes. 

I will do that.

> That behavior could change in the future.

Not likely, this is part of CORBA spec.

From: https://www.omg.org/spec/CORBA/3.3/Interfaces/PDF/changebar

```
7.2.3.1 Escaped Identifiers
As IDL evolves, new keywords that are added to the IDL language may inadvertently collide with identifiers used in
existing IDL and programs that use that IDL. Fixing these collisions will require not only the IDL to be modified, but
programming language code that depends upon that IDL will have to change as well. The language mapping rules for the
renamed IDL identifiers will cause the mapped identifier names (e.g., method names) to be changed.

To minimize the amount of work, users may lexically “escape” identifiers by prepending an underscore (_) to an
identifier. This is a purely lexical convention that ONLY turns off keyword checking. The resulting identifier follows all
the other rules for identifier processing. For example, the identifier _AnIdentifier is treated as if it were AnIdentifier. 
```

**Comment by doudou** [comment]
> Not likely, this is part of CORBA spec.

Oh I missed that one. Still, if it's not too much work, I would prefer the code generated to match the IDL, instead of relying on some obscure (but legal) IDL generator behavior.

**Comment by g-arjones** [comment]
Done: https://github.com/orocos-toolchain/typelib/pull/127

How do you suggest to test this stuff (besides the build tests)?

**Comment by doudou** [comment]
I would add a test in typelib for it. For the orogen bits, to be honest, I think the build test is the best we can do.

File: lib/orogen/marshallers/corba.rb [hunk 1]
```diff
@@ -124,13 +124,24 @@ def corba_name
...
-                "#{corba_namespace}::#{normalize_cxxname(basename.gsub(/[^\w]/, '_'))}_m"
+                "#{corba_namespace}::#{normalize_idl_name(basename)}_m"
             else
-                "#{corba_namespace}::#{normalize_cxxname(basename.gsub(/[^\w]/, '_'))}"
+                "#{corba_namespace}::#{normalize_idl_name(basename)}"
             end
         end
+
+        def strip_leading_underscore(name)
+            name.split("::", -1).map do |field|
```
Comment (doudou) [inline lib/orogen/marshallers/corba.rb:134]: today I learned `split( ..., -1)`

**Comment by doudou** [comment]
Since you say these changes are enough to fix the problem, my suggestion is to merge this and the rock-build-test changes while you add the typelib tests.

**Comment by g-arjones** [comment]
Great. Really appreciate your help on this one. Thanks! :+1:

**Pull request merged**